### PR TITLE
fix: make sure `tar` reproducibility flags are set

### PR DESCRIPTION
**Pull request by drupol**
Related to https://github.com/NixOS/nixpkgs/pull/107251

# Motivation

# Context

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

**Comment by drupol** [comment]
Hi,

This might not be directly needed in `nix`, but this is apparently required to fix this PR in `nixpkgs`: https://github.com/NixOS/nixpkgs/pull/107251

**Review by iFreilicht — APPROVED** [review]
TL:DR; Confirmed, this PR unblocks https://github.com/NixOS/nixpkgs/pull/107251. LGTM!

After *hours* of re-compiling the entire boostrapping chain to compile Nix itself, I've finally reached this:

```
$ nix build .#nixVersions.unstable 
warning: Git tree '/Users/feuh/repos/nixpkgs' is dirty
error: builder for '/nix/store/qaczl05sfdawfaank0dplf1acgdgw87m-nix-2.17.1.drv' failed with exit code 2;
       last 10 log lines:
       >     +(tarball.sh:40) [[ 1689663323 = 1000000000 ]]
       > make: *** [mk/lib.mk:125: tests/tarball.sh.test] Error 1
       > make: *** Waiting for unfinished jobs....
       > ran test tests/simple.sh... [PASS]
       [...]
 ```
 
 on [prusnak/nixpkgs/8f31dc51e](https://github.com/prusnak/nixpkgs/commit/8f31dc51e5ca7a6dc5ee57c9934952da89cff9ff), which is the current HEAD from the PR you linked. So I confirmed the issue was reproducible.
 
I then added an additional version of nix that refers to [drupol/nix/7f71fc754](https://github.com/drupol/nix/commit/7f71fc7540502295202b075f82e89fcf993e7e3a), which is the current HEAD of *this* PR (in [pkgs/tools/package-management/nix/default.nix](
https://github.com/prusnak/nixpkgs/blob/8f31dc51e5ca7a6dc5ee57c9934952da89cff9ff/pkgs/tools/package-management/nix/default.nix#L198)):

```nix
  nix_test_tarball = common rec {
    version = "2.20.0-pre";
    hash = "sha256-T1aqdFg73UB1GjbeSE+iDQkku2/fNS8Fph07s4M570M=";
    src = (fetchFromGitHub { owner = "drupol"; repo = "nix"; rev = "7f71fc7540502295202b075f82e89fcf993e7e3a"; inherit hash; });
  };
```
 
and built that as well:

```
$ nix build .#nixVersions.nix_test_tarball
warning: Git tree '/Users/feuh/repos/nixpkgs' is dirty
```

This time, there was no error. So this PR does exactly what it says.

**Comment by drupol** [comment]
Cool, thanks!
